torch/_inductor/codecache.py

```diff
@@ -321,27 +333,94 @@ def pick_vec_isa():
 
 
 def get_shared(shared=True):
-    return "-shared -fPIC" if shared else ""
+    return " -shared -fPIC" if shared else ""
 
 
 def get_warning_all_flag(warning_all=True):
-    return "-Wall" if warning_all else ""
+    return " -Wall" if warning_all else ""
 
 
 def cpp_flags():
-    return "-std=c++17 -Wno-unused-variable"
+    return " -std=c++17 -Wno-unused-variable"
+
+
+@functools.lru_cache(None)
+def is_omp_valid():
+    def _opt_flags_getter_fn():
+        return "-fopenmp"
 
+    _omp_code = """
+#include <omp.h>
 
+#define N_ELEMENTS (32)
+float in_out_ptr0[N_ELEMENTS] = {0.0};
+
+extern "C" void __avx_chk_kernel() {
+    #pragma omp parallel num_threads(2) {
+        #pragma omp for
+        for(long i0=0; i0<N_ELEMENTS; i0+=1) {
+            in_out_ptr0[i0] = 1.0;
+        }
+}
```

Comment:
We have a cpp_prefix.h - can this live there?

Reply:
Since these are just for compilation checks while `cpp_prefix.h` is meant to be included at runtime. So leaving them (or maybe putting them to a separate header file) is more appropriate?